internal/olm/client/client.go

```diff
@@ -248,25 +265,39 @@ func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedN
 		}
 		depSelectors := ds.Spec.Selector
 		if err := c.KubeClient.Get(ctx, depKey, dep); err != nil {
-			log.Printf("error getting operator deployment %q: %v", ds.Name, err)
+			depErr = append(depErr, deploymentError{
+				name:  ds.Name,
+				issue: err.Error(),
+			})
 			continue
 		}
 		for _, s := range dep.Status.Conditions {
-			if s.Type == appsv1.DeploymentAvailable && s.Status == corev1.ConditionFalse {
-				log.Printf("operator deployment %q not available: %s", ds.Name, s.Reason)
-				if err := c.printPodErrors(ctx, depSelectors, key); err != nil {
-					return err
+			if s.Type == appsv1.DeploymentAvailable && s.Status != corev1.ConditionTrue {
+				podError := c.checkPodErrors(ctx, depSelectors, key)
+				if podError.Error() == "" {
+					depErr = append(depErr, deploymentError{
+						name:  ds.Name,
+						issue: s.Reason,
+					})
+					return depErr
```

Comment:
Other deployments may have errors
```suggestion
					break
```